src/interactive/repl.ts

```diff
@@ -376,6 +376,97 @@ function clearProgress() {
     }
 }
 
+function display(params: { kind: string, data: any }) {
+    if (params.kind === 'application/vnd.julia-vscode.diagnostics') {
+        displayDiagnostics(params)
+    } else {
+        plots.displayPlot(params)
+    }
+}
+
+interface diagnosticData {
+    msg: string,
+    path: string,
+    line?: number,
+    range?: number[][],
+    severity: number,
+    relatedInformation?: {
+        msg: string,
+        path: string,
+        line?: number,
+        range?: number[][]
+    }[]
+}
+const g_trace_diagnostics: Map<string, vscode.DiagnosticCollection> = new Map()
+function displayDiagnostics(params: { kind: string, data: { source: string, items: diagnosticData[] } }) {
```

Comment:
Wouldn't it make more sense to just pass the `data` part of `params` to this function?

Reply:
Ye.

Reply:
Still want to do this?

Reply:
Yes :)